From 06e6af78e741d0c4cfb7a0e4c1b6c7e0207b2681 Mon Sep 17 00:00:00 2001
From: Ethan Chen <intervigil@gmail.com>
Date: Wed, 28 Oct 2015 16:30:51 -0700
Subject: [PATCH 18/19] camera: Allow devices to load custom CameraParameter
 code

* Some devices need additional code to load their cameras. Add a hook
  for extra classes and symbols to be included.

Change-Id: Ifbe79e25b7ab47c5d468f5179032a4283b3f0df5
(cherry picked from commit 38681625992c0029d32eec98fc7e89d71b855b0e)
---
 camera/Android.mk | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/camera/Android.mk b/camera/Android.mk
index 471cb0d..36f6da1 100644
--- a/camera/Android.mk
+++ b/camera/Android.mk
@@ -21,7 +21,6 @@ LOCAL_PATH := $(CAMERA_CLIENT_LOCAL_PATH)
 LOCAL_SRC_FILES:= \
 	Camera.cpp \
 	CameraMetadata.cpp \
-	CameraParameters.cpp \
 	CaptureResult.cpp \
 	CameraParameters2.cpp \
 	ICamera.cpp \
@@ -53,6 +52,21 @@ LOCAL_C_INCLUDES += \
 	system/media/camera/include \
 	system/media/private/camera/include \
 
+ifneq ($(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY),)
+LOCAL_WHOLE_STATIC_LIBRARIES += $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY)
+else
+LOCAL_WHOLE_STATIC_LIBRARIES += libcamera_parameters
+endif
+
 LOCAL_MODULE:= libcamera_client
 
 include $(BUILD_SHARED_LIBRARY)
+
+include $(CLEAR_VARS)
+
+LOCAL_SRC_FILES := \
+	CameraParameters.cpp
+
+LOCAL_MODULE := libcamera_parameters
+
+include $(BUILD_STATIC_LIBRARY)
-- 
2.5.0

